test/unit/log.provider.test.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { sinon } from '@loopback/testlab';
+import { Context } from '@loopback/context';
+import { ParsedRequest } from '@loopback/core';
+import { ExtensionStarterBindings, LogProvider } from '../../src';
+
+describe('LogProvider', () => {
+  it('logs a value without a start time', () => {
+    const match = '/test : () => test message';
+
+    runTest(match);
+  });
+
+  it('logs a value with a start time', () => {
+    const match = 'ms: /test : () => test message';
+    const startTime = process.hrtime();
+
+    runTest(match, startTime);
+  });
+});
+
+function runTest(match: string, startTime?: [number, number]) {
+  const ctx: Context = new Context();
+  ctx.bind(ExtensionStarterBindings.LOG_ACTION).toProvider(LogProvider);
+
+  const log = ctx.getSync(ExtensionStarterBindings.LOG_ACTION);
```

Comment:
Unit test should instantiate the classes directly, they should not use `Context`. Also notice that `log` is of type `any` here, which is not type safe.

I think you should do something like the following:

```ts
const log = new LogProvider().value();
```